### ISSUE TITLE
Training problem of DBnet

### ISSUE THREAD

**Issue by kerisbor** [open-mmlab/mmocr#152]
**Describe the issue**

When I use the default DBnet (res50) config and Synthtext pretrained model to train on ICDAR 2015 , 600 epochs have been run but the recall&precision&hmean is still 0.Is this normal?

**Reproduction**

1. What command or script did you run?

```
./tools/dist_train.sh configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py result 1
```

2. What config dir you run?

```
configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py
```

3. Did you make any modifications on the code or config? Did you understand what you have modified?

4. What dataset did you use?

**Environment**

2021-05-04 16:12:05,375 - mmocr - INFO - Environment info:
------------------------------------------------------------
sys.platform: linux
Python: 3.7.10 (default, Feb 26 2021, 18:47:35) [GCC 7.3.0]
CUDA available: True
GPU 0: TITAN Xp
CUDA_HOME: None
GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
PyTorch: 1.6.0+cu101
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.5.0 (Git Hash e2ac1fac44c5078ca927cb9b90e1b3066a0b2ed0)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.7.0+cu101
OpenCV: 4.5.1
MMCV: 1.2.6
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 10.1
MMOCR: 0.1.0+
**Results**

This is part of the log

```

{"mode": "train", "epoch": 600, "iter": 105, "lr": 0.00376, "memory": 7369, "data_time": 0.11738, "loss_prob": 2.36591, "loss_db": 0.45987, "loss_thr": 0.86378, "loss": 3.68956, "time": 0.86047}
{"mode": "train", "epoch": 600, "iter": 110, "lr": 0.00376, "memory": 7369, "data_time": 0.09441, "loss_prob": 2.18832, "loss_db": 0.39686, "loss_thr": 0.83618, "loss": 3.42136, "time": 0.81025}
{"mode": "train", "epoch": 600, "iter": 115, "lr": 0.00376, "memory": 7369, "data_time": 0.10978, "loss_prob": 2.26413, "loss_db": 0.41894, "loss_thr": 0.83428, "loss": 3.51735, "time": 0.83004}
{"mode": "train", "epoch": 600, "iter": 120, "lr": 0.00376, "memory": 7369, "data_time": 0.11265, "loss_prob": 2.37668, "loss_db": 0.45051, "loss_thr": 0.85281, "loss": 3.68, "time": 0.80677}
{"mode": "train", "epoch": 600, "iter": 125, "lr": 0.00376, "memory": 7369, "data_time": 0.08292, "loss_prob": 2.2562, "loss_db": 0.43429, "loss_thr": 0.80648, "loss": 3.49698, "time": 0.82744}
{"mode": "val", "epoch": 600, "iter": 125, "lr": 0.00376, "hmean-iou:recall": 0.0, "hmean-iou:precision": 0.0, "hmean-iou:hmean": 0.0}
{"mode": "train", "epoch": 601, "iter": 5, "lr": 0.00375, "memory": 7369, "data_time": 0.87823, "loss_prob": 2.49251, "loss_db": 0.53788, "loss_thr": 0.88941, "loss": 3.91981, "time": 1.6014}
{"mode": "train", "epoch": 601, "iter": 10, "lr": 0.00375, "memory": 7369, "data_time": 0.08935, "loss_prob": 2.36297, "loss_db": 0.48122, "loss_thr": 0.85311, "loss": 3.69729, "time": 0.83221}
{"mode": "train", "epoch": 601, "iter": 15, "lr": 0.00375, "memory": 7369, "data_time": 0.08082, "loss_prob": 2.19761, "loss_db": 0.39541, "loss_thr": 0.79208, "loss": 3.38511, "time": 0.79583}
{"mode": "train", "epoch": 601, "iter": 20, "lr": 0.00375, "memory": 7369, "data_time": 0.11728, "loss_prob": 2.4965, "loss_db": 0.46152, "loss_thr": 0.84559, "loss": 3.80361, "time": 0.84817}
{"mode": "train", "epoch": 601, "iter": 25, "lr": 0.00375, "memory": 7369, "data_time": 0.12661, "loss_prob": 2.39774, "loss_db": 0.49221, "loss_thr": 0.87436, "loss": 3.76431, "time": 0.85035}
{"mode": "train", "epoch": 601, "iter": 30, "lr": 0.00375, "memory": 7369, "data_time": 0.08912, "loss_prob": 2.43385, "loss_db": 0.51091, "loss_thr": 0.84726, "loss": 3.79202, "time": 0.81482}
{"mode": "train", "epoch": 601, "iter": 35, "lr": 0.00375, "memory": 7369, "data_time": 0.1037, "loss_prob": 2.28464, "loss_db": 0.46282, "loss_thr": 0.85413, "loss": 3.60159, "time": 0.82637}
{"mode": "train", "epoch": 601, "iter": 40, "lr": 0.00375, "memory": 7369, "data_time": 0.11033, "loss_prob": 2.12298, "loss_db": 0.38375, "loss_thr": 0.77541, "loss": 3.28214, "time": 0.82024}
{"mode": "train", "epoch": 601, "iter": 45, "lr": 0.00375, "memory": 7369, "data_time": 0.11287, "loss_prob": 2.13396, "loss_db": 0.37305, "loss_thr": 0.79603, "loss": 3.30303, "time": 0.81564}

```

**Comment by kerisbor** [comment]
the log of DBnet(res50) is not available.
https://github.com/open-mmlab/mmocr/tree/main/configs/textdet/dbnet#icdar2015

**Comment by jeffreykuang** [comment]
[dbnet_r50dcnv2_fpnc_sbn_1200e_icdar2015_20210325-91cef9af.log](https://github.com/open-mmlab/mmocr/files/6431246/dbnet_r50dcnv2_fpnc_sbn_1200e_icdar2015_20210325-91cef9af.log)

attached is my training log. pls compare my log with yours. @cuhk-hbsun  pls upload it to the download sever and check the link is correct or not.

To figure out non-convergence,  I will try to recover your issue via retraining the model using the latest code and config.

BTW, during reimplementing dbnet, I found that bbce_loss is not easy to converge when BN layers are fixed. Replacing bbce_loss by dice loss will converge very fast although this is not the same as reported in the original paper. You can replace bbce_loss by dice_loss via setting  bbce_loss=False at https://github.com/open-mmlab/mmocr/blob/0e70f99f4d8fa0180bdab16f8697e65e71590c72/configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py#L26

**Comment by kerisbor** [comment]
> [dbnet_r50dcnv2_fpnc_sbn_1200e_icdar2015_20210325-91cef9af.log](https://github.com/open-mmlab/mmocr/files/6431246/dbnet_r50dcnv2_fpnc_sbn_1200e_icdar2015_20210325-91cef9af.log)
> 
> attached is my training log. pls compare my log with yours. @cuhk-hbsun pls upload it to the download sever and check the link is correct or not.
> 
> To figure out non-convergence, I will try to recover your issue via retraining the model using the latest code and config.
> 
> BTW, during reimplementing dbnet, I found that bbce_loss was not easy to converge when BN layers are fixed. However, it does converge when BN layers are not fixed during training in my experiments. Another observation is that Replacing bbce_loss by dice loss will converge very fast although this is not the same as reported in the original paper. You can replace bbce_loss by dice_loss via setting bbce_loss=False at
> 
> https://github.com/open-mmlab/mmocr/blob/0e70f99f4d8fa0180bdab16f8697e65e71590c72/configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py#L26

I used the latest code and now I can train normally. Thank you very much for your reply !

**Comment by jeffreykuang** [comment]
@cuhk-hbsun  pls upload the log before closing this issue

**Comment by cuhk-hbsun** [comment]
fixed
